src/core/Akka.Remote/Transport/AkkaPduCodec.cs

```diff
@@ -202,12 +202,12 @@ public AckAndMessage(Ack ackOption, Message messageOption)
     internal abstract class AkkaPduCodec
     {
         protected readonly ActorSystem System;
-        protected readonly AddressThreadLocalCache AddressCache;
+        protected readonly ActorPathThreadLocalCache ActorPathCache;
 
         protected AkkaPduCodec(ActorSystem system)
         {
             System = system;
-            AddressCache = AddressThreadLocalCache.For(system);
+            ActorPathCache = ActorPathThreadLocalCache.For(system);
```

Comment:
Yes, AddressCache is now dark

The AkkaPduCodec (better the EndPointReader) is the best place to directly instanced the ActorPathCache and use it.
When then ActorPathCache is freed from ThreadLocal and bound to an Endpoint then it would produce the best cache-hits.

The only thing what needs an API change, would be the `provider.ResolveActorRefWithLocalAddress(envelopeContainer.Recipient.Path, localAddress)`  call.
this needs to be extended or replaced with an Context parameter.
Something like `ResolveContext` with LocalAddress and the ActorPathCache

Reply:
Not sure I understand that fully just yet but if that's part of a future PR we can look at it then

Reply:
The ActorPaths are strongly cupelled to the RemoteNode to have a single ActorPathCache in each EndPointReader
instead of a shared over ThreadLocal, would very increase the hit-rate. 

What I saw, only the read-side needs the ActorPathCache